rest/data/generate.go

```diff
@@ -51,10 +63,29 @@ func (gc *GenerateConnector) GeneratePoll(ctx context.Context, taskID string, gr
 	if err != nil {
 		return false, nil, errors.Wrapf(err, "problem getting queue for version %s", t.Version)
 	}
-	jobID := fmt.Sprintf("generate-tasks-%s", taskID)
-	j, exists := q.Get(ctx, jobID)
-	if !exists {
-		return false, nil, errors.Errorf("task %s not in queue", taskID)
+
+	var jobID string
+	var j amboy.Job
+	var exists bool
+	for {
```

Comment:
check context cancellation in this loop?

Reply:
done